x-pack/filebeat/input/cloudfoundry/main_test.go

```diff
@@ -0,0 +1,28 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cloudfoundry
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/elastic/beats/v7/libbeat/paths"
+)
+
+func TestMain(m *testing.M) {
```

Comment:
`TestMain` is part of the `testing` package, it defines how tests are run for a package: https://golang.org/pkg/testing/#hdr-Main.
Here I am configuring the data directory to avoid creating directories in the working copy on testing.
Maybe we should have another way of configuring the data directory on testing.

Reply:
oh my, I'm blind 🙈 ! sorry for the noise.

Yes, I understand this may already be an issue with other files like meta or the registry?

Reply:
> Yes, I understand this may already be an issue with other files like meta or the registry?

Yes, this is not the first time I find issues with tests and the data directory.